src/rust/ensogl/lib/core/src/display/scene/dom.rs

```diff
@@ -193,18 +191,21 @@ impl DomScene {
     pub fn update_view_projection(&self, camera:&Camera2d) {
         if self.children_number() == 0 { return }
 
-        let trans_cam  = camera.transform_matrix().try_inverse();
-        let trans_cam  = trans_cam.expect("Camera's matrix is not invertible.");
+        let trans_cam  = camera.view_matrix();
         let trans_cam  = trans_cam.map(eps);
         let trans_cam  = inverse_y_translation(trans_cam);
-        let half_dim   = camera.screen().height / 2.0;
-        let fovy_slope = camera.half_fovy_slope();
-        let near       = half_dim / fovy_slope;
+        let zoom = camera.zoom();
```

Comment:
I'm not sure this works as expected now. The lines you have removed were re-computing the zoom after for example scene dimension change. So, when you are resizing the window, you want objects to be exactly the same size they were earlier, so camera needs to recompute its zoom. This needs to be pixel perfect, without even 0.5 px transformation in any direction, because it will cause blurred renders in EnsoGL vector renderer. These equations were designed to recompute zoom with high precision